include/bitcoin/system/wallet/witness_address.hpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2011-2019 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBBITCOIN_SYSTEM_WALLET_WITNESS_ADDRESS_HPP
+#define LIBBITCOIN_SYSTEM_WALLET_WITNESS_ADDRESS_HPP
+
+#include <bitcoin/system/math/hash.hpp>
+#include <bitcoin/system/wallet/payment_address.hpp>
+
+namespace libbitcoin {
+namespace system {
+namespace wallet {
+
+static BC_CONSTEXPR uint8_t padding = 0x00;
+static BC_CONSTEXPR size_t witness_p2wpkh_size = 3u + short_hash_size + checksum_size;
+static BC_CONSTEXPR size_t witness_p2wsh_size = 3u + hash_size + checksum_size;
```

Comment:
Declare `witness_prefix_size = 3u`. Should update other classes as well so that magic numbers are self-documenting.